### PR TITLE
Added ability to work with orders

### PR DESCRIPTION
**Pull request by HovorukhaBohdan**
`POST: "/api/orders"` - place orders 
`GET: "/api/orders"` - check orders history
`GET: "/api/orders/{orderId}/items"` - check items in specific orders
`GET: "/api/orders/{orderId}/items/{itemId}"` - check specific item in specific orders
`PATCH:  "/api/orders/{id}"` - update status for order

### CODE REVIEW

File: src/main/java/com/example/bookshop/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        return orderService.getOrdersHistory(user.getId(), pageable);
+    }
+
+    @ResponseStatus(HttpStatus.CREATED)
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PostMapping
+    @Operation(summary = "Place an order", description = "Place an order")
+    public OrderResponseDto placeOrder(
+            Authentication authentication,
+            Pageable pageable,
```
Comment (d1sam) [inline src/main/java/com/example/bookshop/controller/OrderController.java]: why do you need `Pageable`? pls remove :)

File: src/main/java/com/example/bookshop/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    ) {
+        return orderService.updateOrderStatus(id, requestDto);
+    }
+
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/{orderId}/items")
+    @Operation(summary = "Get all item from a certain order",
+            description = "Get all item from a certain order")
+    public List<OrderItemResponseDto> getAllOrderItemsForSpecificOrder(
+            Authentication authentication,
```
Comment (d1sam) [inline src/main/java/com/example/bookshop/controller/OrderController.java]:
```suggestion
            Authentication authentication,
            Pageable,
```

File: src/main/java/com/example/bookshop/dto/orderitem/OrderItemRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.example.bookshop.dto.orderitem;
+
+import jakarta.validation.constraints.NotEmpty;
+import lombok.Data;
+
+@Data
+public class OrderItemRequestDto {
+    @NotEmpty(message = "can't be empty")
```
Comment (d1sam) [inline src/main/java/com/example/bookshop/dto/orderitem/OrderItemRequestDto.java]:
```suggestion
    @NotBlank(message = "can't be empty")
```

File: src/main/java/com/example/bookshop/model/Order.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    private BigDecimal total;
+
+    @Column(name = "order_date", nullable = false)
+    private LocalDateTime orderDate;
+
+    @Column(name = "shipping_address", nullable = false)
+    private String shippingAddress;
+
+    @OneToMany(mappedBy = "order")
+    private Set<OrderItem> orderItems;
```
Comment (d1sam) [inline src/main/java/com/example/bookshop/model/Order.java:53]:
```suggestion
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private Set<OrderItem> orderItems;
```

File: src/main/java/com/example/bookshop/repository/OrderRepository.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface OrderRepository extends JpaRepository<Order, Long> {
+    @Query("SELECT o FROM Order o JOIN FETCH o.orderItems WHERE o.user.id = :userId")
+    List<Order> findAllByUserId(Long userId, Pageable pageable);
+
+    @EntityGraph(attributePaths = {"orderItems"})
+    @NonNull
```
Comment (d1sam) [inline src/main/java/com/example/bookshop/repository/OrderRepository.java]:
```suggestion
```

File: src/main/java/com/example/bookshop/service/impl/OrderServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+    @Override
+    public List<OrderResponseDto> getOrdersHistory(Long userId, Pageable pageable) {
+        List<Order> ordersHistory = orderRepository.findAllByUserId(userId, pageable);
+        return ordersHistory.stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
```
Comment (d1sam) [inline src/main/java/com/example/bookshop/service/impl/OrderServiceImpl.java:45]:
```suggestion
    @Override
    @Transactional
```

**Review by d1sam — CHANGES_REQUESTED** [review]
good job, but some changes should be done!

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**